source/validate.cpp

```diff
@@ -279,6 +475,19 @@ spv_result_t spvValidate(const spv_const_context context,
     return SPV_ERROR_INVALID_BINARY;
   }
 
+  ValidationState_t vstate(pDiagnostic);
```

Comment:
It's not clear what the evolution of the algorithm is going to be.
I suspect you're going to do per-instruction processing in a single spvBinaryParse, but also collect info to be checked after the fact.  Should give that high level view here.